lnd_test.go

```diff
@@ -5307,13 +5320,158 @@ func testBasicChannelCreation(net *lntest.NetworkHarness, t *harnessTest) {
 				Amt: amount,
 			},
 		)
+
+		// Since the channel just became open, we should receive an active
+		// notification.
+		activeChannelUpdate, err := bobChannelSubscription.Recv()
+		if err != nil {
+			t.Fatalf("unable to recv channel update: %v", err)
+		}
+
+		switch chanUpdate := activeChannelUpdate.Channel.(type) {
+		case *lnrpc.ChannelUpdate_OpenChannel:
+			if !chanUpdate.OpenChannel.Active ||
+				activeChannelUpdate.Type != lnrpc.ChannelUpdate_ACTIVE_CHANNEL {
+				t.Fatalf("channel not active but should be")
+			}
+		default:
+			t.Fatalf("channel update channel of wrong type, expected OpenChannel,"+
+				"got: %T", chanUpdate)
+		}
 	}
 
-	// Close the channel between Alice and Bob, asserting that the
-	// channel has been properly closed on-chain.
-	for _, chanPoint := range chanPoints {
+	// Subscribe Alice to channel updates so we can test that both remote
+	// and local force close notifications are received correctly.
+	aliceReq := &lnrpc.ChannelSubscription{}
+	ctx, cancelAliceSubscription := context.WithCancel(context.Background())
+	defer cancelAliceSubscription()
+
+	aliceChannelSubscription, err := net.Alice.SubscribeChannels(ctx, aliceReq)
+	if err != nil {
+		t.Fatalf("unable to subscribe to alice's channel updates: %v", err)
+	}
+
+	// Close the channel between Alice and Bob, asserting that the channel
+	// has been properly closed on-chain and ensuring that the correct
+	// notifications are received.
+	for i, chanPoint := range chanPoints {
 		ctx, _ := context.WithTimeout(context.Background(), timeout)
-		closeChannelAndAssert(ctx, t, net, net.Alice, chanPoint, false)
+
+		// Force close half of the channels.
+		var force bool
+		switch {
+		case i%1 == 0:
+			force = true
+		default:
+			force = false
+		}
+		closeChannelAndAssert(ctx, t, net, net.Alice, chanPoint, force)
+
+		// Since the channel just became inactive, we should receive an
+		// inactive notification from each subscription.
+		inactiveChannelUpdate, err := bobChannelSubscription.Recv()
```

Comment:
Do we timeout here if we don't receive a notification? If not, these calls should be extracted into helper methods that let us set a timeout (I think we do something similar for graph updates).